src/Symfony/Component/Messenger/Envelope.php

```diff
@@ -40,6 +41,19 @@ public function __construct($message, StampInterface ...$stamps)
         }
     }
 
+    /**
+     * Makes sure the message is in an Envelope and adds the given stamps.
+     *
+     * @param object|Envelope  $message
+     * @param StampInterface[] $stamps
+     */
+    public static function wrap($message, array $stamps = []): self
```

Comment:
@weaverryan wouldnt stamp() be the better name? In case of explicit public usage ...

Reply:
to clarify; if wrapping means "put it in an envelop" this method doesnt always "wrap", whereas `new Envelope($anything)` does.

This method wraps if needed, but always stamps.

Reply:
So, are you suggesting a change? Or is it clear for you now?

Reply:
It stamps only if stamps are provided.
The main use case is removing the "if" that the method contains from userland boilerplate.
Tldr, wrap() fits my mind :)

Reply:
yeah it might fit; "we wrap it with stamps" :/ but to me that's "stamping", not "wrapping".

Just curious if wrap() and __construct() could be considered confusing...

Reply:
> It stamps only if stamps are provided.

hm indeed, missed this detail :}  OK this is technical details are hard to convey with naming :) let's keep as is :+1: